app/src/main/java/mozilla/lockbox/action/TelemetryAction.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.action
+
+import mozilla.lockbox.flux.Action
+import org.mozilla.telemetry.event.TelemetryEvent
+
+open class TelemetryAction(
+    val eventMethod: TelemetryEventMethod,
+    val eventObject: TelemetryEventObject,
+    val value: String?,
+    val extras: Map<String, Any>?
+) : Action {
+    open fun createEvent(category: String = "action"): TelemetryEvent {
+        val evt = TelemetryEvent.create(
+                category,
+                eventMethod.name,
+                eventObject.name,
+                value
+        )
+        extras?.forEach { ex -> evt.extra(ex.key, ex.value.toString()) }
+
+        return evt
+    }
+}
+
+enum class TelemetryEventMethod {
+    tap,
+    startup,
+    foreground,
+    background,
+    settingChanged,
+    show,
+    canceled,
+    login_selected,
+    autofill_locked,
+    autofill_unlocked,
+    refresh,
+    autofill_clear
+}
+
+enum class TelemetryEventObject {
```

Comment:
🎉 way to get these all in early! follow-up stories will be supa easy ^.^

Reply:
Also the follow up stories should already exist.